### ISSUE TITLE
Make it clearer that Out-File uses the formatting system, contrast it with Set-Content

### ISSUE THREAD

**Issue by mklement0** [MicrosoftDocs/PowerShell-Docs#6600]
As an aside: Currently, the "Description" section reads:

> The `Out-File` cmdlet sends output to a file. When you need to specify parameters for the output use `Out-File` rather than the redirection operator (`>).

Also worth mentioning that `>>` is functionally equivalent to `Out-File -Append`.

---

While the "Notes" section talks about formatting, I think it is important to provide this information up front, in the "Description" section:

* `Out-File` implicitly uses PowerShell's formatting system, which means the file will receive _the same for-display representation that you would see in the console_ (terminal).

  * As such, files created this way are _not_ generally suited for programmatic processing - unless all input objects are _strings_. 

* By contrast, `Set-Content` does _not_ use the formatting system and stringifies input objects on demand by calling their `.ToString()` method.

Since both cmdlets use potentially _culture-sensitive_ representations, files created by them are only suitable for robust programmatic processing if the input objects are deliberately constructed _strings_ that use _culture-invariant_ representations; piping from cmdlets such as `ConvertTo-Json` or `ConvertTo-Csv` are examples.

`Export-CliXml` is the most robust choice for storing _typed_ data in files, though even there the type fidelity on reading (deserialization with `Import-CliXml`) has limits.

---

The "Notes" section currently reads:

> The Out cmdlets do not format objects; they just render them and send them to the specified display destination. If you send an unformatted object to an Out cmdlet, the cmdlet sends it to a formatting cmdlet before rendering it.

I find this wording a bit confusing, because from the user's perspective `Out-File` _does_ perform formatting - implicitly, just like in the console (which you can think of as piping to `Out-Host`).

To put it differently: If you do not send output from a `Format-*`cmdlet explicitly to `Out-File`, `Out-File` automatically picks the output-type appropriate `Format-*` cmdlet to use behind the scenes, as it does in the console.

> Out-File sends data but it does not produce any output objects. If you pipe the output of Out-File to Get-Member, the Get-Member cmdlet reports that no objects were specified.

I think this explanation is ultimately more confusing than helpful.
All the user needs to know is that `Out-File` produces no output to the _pipeline_ - it only saves data to a _file_.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: cf24acea-6ecc-ca38-f183-3a71c76d6c84
* Version Independent ID: 6ca0aa0a-2b48-273f-cd95-831d08bef772
* Content: [Out-File (Microsoft.PowerShell.Utility) - PowerShell](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/out-file?view=powershell-7)
* Content Source: [reference/7.0/Microsoft.PowerShell.Utility/Out-File.md](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/7.0/Microsoft.PowerShell.Utility/Out-File.md)
* Product: **powershell**
* Technology: **powershell-cmdlets**
* GitHub Login: @JamesWTruher
* Microsoft Alias: **jimtru**

**Comment by chasewilson** [comment]
- I didn't exlicitly add the information about the redirection operator since that is in fact something separate. However, I do think that the redirection should be referenced and linked to so they can get complete information about it. 

- I agree with the first part of this section and am making these changes. I think that adding information about the contrasting behavior, especially since stringification has it's added complexity when working with complex data types, I think adding information will only add unessary confusion and scope creep a bit. 

- Tried to just simplify this section here. I think it's worth noting that it doesn't format it itself but sends it to a formatter. Let me know what you think. I basically just trimmed the section. I also just tried to simplify the wording on the next point as well to avoid unneccessary complexity.

**Comment by mklement0** [comment]
Thanks, @chasewilson, I'll take a look at the PR.

As for:

> I didn't explicitly add the information about the redirection operator since that is in fact something separate.

Yes, `>` isn't an _alias_ of `Out-File`, but (a) it isn't separate in terms of _user experience_ (which is what matters most) and it also isn't separate in terms of _implementation_, as evidenced by the fact that you can change the encoding used by `>` via a `$PSDefaultParameterValues` entry targeting `Out-File` - the latter may in itself be worth mentioning: 

```powershell
# Note the UTF-16LE output produced by `>`
PS> $PSDefaultParameterValues['Out-File:Encoding'] = 'unicode'; 'hi'>t.txt; fhx t.txt
          Offset Bytes                                           Ascii
                 00 01 02 03 04 05 06 07 08 09 0A 0B 0C 0D 0E 0F
          ------ ----------------------------------------------- -----
0000000000000000 FF FE 68 00 69 00 0A 00                         ÿþh i �
```
